includes/api/class-wc-admin-rest-admin-notes-controller.php

```diff
@@ -0,0 +1,304 @@
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'system_status', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'wc-admin' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Prepare a note object for serialization.
+	 *
+	 * @param array           $data Note data.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $data, $request ) {
+		$context                   = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data                      = $this->add_additional_fields_to_object( $data, $request );
+		$data['date_created_gmt']  = wc_rest_prepare_date_response( $data['date_created'] );
+		$data['date_created']      = wc_rest_prepare_date_response( $data['date_created'], false );
+		$data['date_reminder_gmt'] = wc_rest_prepare_date_response( $data['date_reminder'] );
+		$data['date_reminder']     = wc_rest_prepare_date_response( $data['date_reminder'], false );
+		$data                      = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+		$response->add_links( array(
+			'self'       => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d', $this->namespace, $this->rest_base, $data['id'] ) ),
+			),
+			'collection' => array(
+				'href' => rest_url( sprintf( '%s/%s', $this->namespace, $this->rest_base ) ),
+			),
+		) );
+		/**
+		 * Filter a note returned from the API.
+		 *
+		 * Allows modification of the note data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param array            $data The original note.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_admin_note', $response, $data, $request );
+	}
+
+	/**
+	 * Retrieves the item's schema for display / public consumption purposes.
+	 *
+	 * @return array Public item schema data.
+	 */
+	public function get_public_item_schema() {
+		$schema = parent::get_public_item_schema();
+
+		$schema['properties']['id'] = array(
+			'description' => __( 'ID of the note record.', 'wc-admin' ),
+			'type'        => 'integer',
+			'context'     => array( 'view', 'edit' ),
+			'readonly'    => true,
+		);
+
+		$schema['properties']['name'] = array(
+			'description' => __( 'Name of the note.', 'wc-admin' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'readonly'    => true,
+		);
+
+		$schema['properties']['type'] = array(
+			'description' => __( 'Type of the note.', 'wc-admin' ),
```

Comment:
Type of note?

Reply:
Reworded a bit in https://github.com/woocommerce/wc-admin/pull/507/commits/931b45f71d97a3ea5ee15244c68355514e947d95